### PR TITLE
fix!: export `typescript-eslint` instead of `@typescript-eslint/eslint-…

### PR DESCRIPTION
**Pull request by jerome-benoit**
…plugin`

It's actually where:

- complete shareable configurations are available
- configuration helper is available

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by jerome-benoit** [comment]
There's an issue with not allowed reexporting: 

```
ConfigError: Config "neostandard/ts": Key "plugins": Cannot redefine plugin "@typescript-eslint".
    at rethrowConfigError
```

I put it as a draft until I figure a workaround.

**Comment by voxpelli** [comment]
Yeah, this is why I exported the plugin, since that's what's referenced in the neostandard config and that's what needs to be reused, hmm

**Comment by voxpelli** [comment]
@jerome-benoit I believe the reason why I used https://github.com/neostandard/neostandard/blob/05f976994852b71e90619b87d21854f95bc8a17f/lib/ts.js#L3-L4

and not:

```js
const { parser, plugin } = require('typescript-eslint')
```

is that `@typescript-eslint/eslint-plugin` returns a richly typed plugin, whereas `typescript-eslint` simply types it as a generic `TSESLint.FlatConfig.Plugin`, hmm

**Comment by voxpelli** [comment]
@jerome-benoit What do you think about this solution?

**Comment by jerome-benoit** [comment]
> is that `@typescript-eslint/eslint-plugin` returns a richly typed plugin, whereas `typescript-eslint` simply types it as a generic `TSESLint.FlatConfig.Plugin`, hmm

The issue mentioned is also present in the 0.10 release: https://github.com/jerome-benoit/poolifier/actions/runs/9749052301/job/26905401063. As long as typescript-eslint shareable configurations are used, eslint is erroing because of the plugin name key redefinition: module.exports usage seems to trigger it.  

The issue with the  '@typescript-eslint/eslint-plugin' reexport is that it's not really relevant for the shareable configurations: the .configs property does not expose complete reusable configurations.

**Comment by jerome-benoit** [comment]
> @jerome-benoit What do you think about this solution?

The dependencies cleanup looks better, but I need to test that branch against poolifier repo to ensure it fix the error, not sure I will be able tonight.  

I think a test for reexport use should be planned, at least for typescript-eslint one that seems trigger a code path in eslint checking for plugin name change.

**Comment by voxpelli** [comment]
> I think a test for reexport use should be planned, at least for typescript-eslint one that seems to trigger a code path in eslint checking for plugin name change.

Added ✔️ And I did it in a `.mjs` file to ensure the mixture of CJS and ESM did not mess with anything

File: test/ts-extension-eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import neostandard, { plugins } from '../index.js'
+
+export default [
+  ...neostandard({
+    ts: true,
+  }),
+  ...plugins['typescript-eslint'].configs.recommended,
```
Comment (jerome-benoit) [inline test/ts-extension-eslint.config.mjs:7]: I've added a similar test locally: 

```ts
 ...plugins['typescript-eslint'].config(
    ...plugins['typescript-eslint'].configs.recommended,
    ...plugins['typescript-eslint'].configs.strict
  )
```

just to be sure to test the helper with several configurations.

Comment (voxpelli) [reply test/ts-extension-eslint.config.mjs:7]: Should we add such a variant here as well do you think?

**Comment by jerome-benoit** [comment]
Looks good. Since I can't approve a PR I've opened ... :-)

**Review by voxpelli — APPROVED** [review]
Yeah, that's one downside of hijacking a PR like I did now 🙈 Was torn on whether to open a new one, but that felt even more like hijack

File: test/ts-extension-eslint.config.mjs (hunk 1 above)
Comment (jerome-benoit) [reply test/ts-extension-eslint.config.mjs:7]: It depends if the helper is actually triggering the error: on poolifier, if I comment out 

 ...tseslint.config(...) 

there's no error.

**Pull request merged**